app/models/ext_management_system.rb

```diff
@@ -50,23 +46,19 @@ def self.supported_types_for_create
   end
 
   def self.supported_types_for_catalog
-    subclasses_supports?(:catalog)
+    subclasses_supporting(:catalog)
   end
 
   def self.supported_for_create?
     !reflections.include?("parent_manager")
   end
 
-  def self.label_mapping_classes
-    subclasses_supports?(:label_mapping)
-  end
-
   def self.label_mapping_prefixes
-    label_mapping_classes.map(&:label_mapping_prefix).uniq
+    subclasses_supporting(:label_mapping).map(&:label_mapping_prefix).uniq
   end
 
   def self.entities_for_label_mapping
-    label_mapping_classes.reduce({}) { |all_mappings, klass| all_mappings.merge(klass.entities_for_label_mapping) }
+    subclasses_supporting(:label_mapping).reduce({}) { |all_mappings, klass| all_mappings.merge(klass.entities_for_label_mapping) }
```

Comment:
would this work  does it read better?
```suggestion
    subclasses_supporting(:label_mapping).flat_map(&:entities_for_label_mapping)
```

Reply:
Yeah `.flat_map` is probably how I would have written it

Reply:
Those return two different things...one creates a hash and the other creates an array.

Reply:
dumb idea, who would suggest this?

(aka sorry)